network/src/protocols/network/mod.rs

```diff
@@ -172,22 +174,58 @@ pub trait NewNetworkEvents {
     fn new(
         peer_mgr_notifs_rx: aptos_channel::Receiver<(PeerId, ProtocolId), PeerManagerNotification>,
         connection_notifs_rx: aptos_channel::Receiver<PeerId, ConnectionNotification>,
+        bounded_executor: BoundedExecutor,
     ) -> Self;
 }
 
-impl<TMessage: Message> NewNetworkEvents for NetworkEvents<TMessage> {
+impl<TMessage: Message + Send + 'static> NewNetworkEvents for NetworkEvents<TMessage> {
     fn new(
-        peer_mgr_notifs_rx: aptos_channel::Receiver<(PeerId, ProtocolId), PeerManagerNotification>,
+        mut peer_mgr_notifs_rx: aptos_channel::Receiver<
+            (PeerId, ProtocolId),
+            PeerManagerNotification,
+        >,
         connection_notifs_rx: aptos_channel::Receiver<PeerId, ConnectionNotification>,
+        bounded_executor: BoundedExecutor,
```

Comment:
passing the executor here seems causing a lot of changes. does `for_each_concurrent` on the peer_mgr_notifs_rx not work?

Reply:
Yeah, good call. I changed to `for_each_concurrent`. We still need to pass through the max number of deserialization tasks parameter, but, it should hopefully be cleaner now 😄